components/ParameterForm.vue

```diff
@@ -0,0 +1,236 @@
+          :label="globeParameter.label"
+          :aria-label="globeParameter.label"
+          :options="selectOptions(globeParameter)"
+          :size="largeScreen ? 'lg' : ''"
+        />
+      </div>
+      <CButton
+        id="run-button"
+        color="primary"
+        :size="largeScreen ? 'lg' : ''"
+        type="submit"
+      >
+        Run
+        <CIcon
+          icon="cilArrowRight"
+        />
+      </CButton>
+    </CForm>
+    <CAlert v-else-if="props.metadataFetchStatus === 'error'" color="warning">
+      Failed to retrieve metadata from R API. {{ metadataFetchError }}
+    </CAlert>
+    <CSpinner v-else-if="props.metadataFetchStatus === 'pending'" />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import type { FetchError } from "ofetch";
+import { CIcon } from "@coreui/icons-vue";
+import type { MetaData, Parameter, ParameterOption } from "@/types/daedalusApiResponseTypes";
+import type { AsyncDataRequestStatus } from "#app";
+
+const props = defineProps<{
+  globeParameter: Parameter | undefined
+  metaData: MetaData | undefined
+  metadataFetchStatus: AsyncDataRequestStatus
+  metadataFetchError: FetchError | null
+}>();
+
+const formData = ref(
+  // Create a new object with keys set to the id values of the metaData.parameters array of objects, and all values set to refs with default values.
+  props.metaData?.parameters.reduce((accumulator, parameter) => {
+    if (parameter.parameterType !== "select" && parameter.parameterType !== "globeSelect") {
+      accumulator[parameter.id] = ref("");
+      return accumulator;
+    }
+
+    // TODO: Make default country UK after November 2024 workshop
+    if (parameter.id === "country") {
+      accumulator[parameter.id] = ref("Thailand");
+    } else {
+      // Don't set an empty value or there will be a disjoint between the select component and the formData object,
+      // since the select component will visually appear to have an option selected (the first), but the formData object will have
+      // an empty string.
+      const defaultOption = parameter.defaultOption || parameter.options[0].id;
+      accumulator[parameter.id] = ref(defaultOption);
+    }
+
+    return accumulator;
+  }, {} as { [key: string]: any }),
+);
+
+const parametersOfTypeSelect = computed(() => {
+  if (props.metaData) {
+    return props.metaData.parameters.filter(parameter => parameter.parameterType === "select");
+  } else {
+    return [];
+  }
+});
+
+const globeParameter = computed(() => {
+  if (props.metaData) {
+    return props.metaData.parameters.filter(parameter => parameter.parameterType === "globeSelect")[0];
+  } else {
+    return undefined;
+  }
+});
+
+const icon = (parameter: Parameter) => {
+  switch (parameter.id) {
+    case "country":
+      return "cilGlobeAlt";
+    case "response":
+      return "cilShieldAlt";
+    case "vaccine":
+      return "cilIndustry";
+    case "pathogen":
+      return "cilBug";
```

Comment:
> arguably the preferred icon should be provided as part of the metadata too
I'm not sure I see the benefit. If there is an abstraction layer, we'd still end up with the web app having a dict look up.

Reply:
You make a good point about the bug, I wasn't sure myself. I will look for a little virus icon that makes our style.

Reply:
The colour's a little off, making my eye twitch, but I hope people can live with it for now because I couldn't quickly fix it (when you have inline svg you can change a fill attribute, but that doesn't seem to work when it's an svg loaded in an img tag)